### PR TITLE
Updating sans GUI table

### LINKED ISSUE #23022: Update sans GUI to use JobTreeView
The JobTreeView is now finished so should be put into the SANS GUI.

### PR DESCRIPTION
**Pull request by Matthew-Andrew**
**Description of work.**

This is the initial work to replace the DataProcessorWidget with the JobTreeView widget in the new Sans GUI. This has involved doing the following:
* Updating all the methods on the view to interact with the JobTreeView rather than the DataProcessorWidget
* Adding a removeAllRows method to JobTreeView
* Updated the table model to keep an up to date version of the table in the view. 
* In main_presenter.py removed the inheritance from DataProcessorMainPresenter as well as all the now unneeded code. This class is now basically uneeded and will probably be removed in a future branch.
* Updating run_tab_presenter to interact with the new table_view and table_model as well as run the reduction process itself.

**To test:**
* Attempt to run a SANS reduction in the new GUI
* Play around and check usability is good. 

Fixes #23022 . 

Does this update require release notes?
- [ ] Yes
- [x] No

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [x] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [x] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by gemmaguest** [comment]
I get this on trying to open the GUI:
```
AttributeError: 'MainPresenter' object has no attribute 'get_gui_algorithm_name'
  at line 19 in '<Interface>'
  caused by line 131 in '/opt/mantidnightly/scripts/Interface/ui/sans_isis/sans_data_processor_gui.py'
```

**Comment by Mantid-Matthew** [comment]
Hmm


It looks like it is trying to use a file from a nightly build which will still be looking for a load of old stuff.

I've had something similar previously and found I had to clear out the directories I was searching for python files in the properties.user file. This is the one in .mantid on linux or in %appdata% on windows.


Does that do anything ?

________________________________

File: MantidPlot/src/ProjectRecovery.cpp [hunk 1]
```diff
@@ -365,6 +365,7 @@ void ProjectRecovery::loadRecoveryCheckpoint(const Poco::Path &recoveryFolder) {
     this->startProjectSaving();
     return;
   }
+
```
Comment (gemmaguest) [inline MantidPlot/src/ProjectRecovery.cpp]: Was this file supposed to be changed?

**Review by gemmaguest — CHANGES_REQUESTED** [review]
Looks good to me :+1:

Just one file which looks like it's been changed accidentally which would be good to revert to avoid confusion.

Also just to note, the HintingLineEdit is not implemented so the Options column does not show tooltips. It also doesn't reject invalid input. (However, valid input is used correctly.) As this is merging to a feature branch it's fine to add the missing behaviour in a future PR.

**Review by gemmaguest — APPROVED** [review]

**Comment by gemmaguest** [comment]
Happy to merge when tests pass :+1:

**Comment by martyngigg** [comment]
@Mantid-Matthew Was this branch started from `release-next`? It looks like there are some fixes that are on master that are necessary for the RHEL7 builds that are now using gcc 7. I would suggest moving the `sans_table_change_feature_branch` forward to master and then merging that to this branch.

**Comment by Mantid-Matthew** [comment]
@martyngigg Yes that was the problem they were running an old version of gcc on RHEL7 all seems to be good now.

**Pull request merged**